src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -3995,14 +4000,17 @@ private void processFileChanges(
             // This is for server.xml specified by the configuration parameter
             // server will load new properties
             if (fileChanged.exists() && (changeType == ChangeType.MODIFY || changeType == ChangeType.CREATE)) {
-                // suppress install feature warning - property must be set before calling copyConfigFolder
-                System.setProperty(SKIP_BETA_INSTALL_WARNING, Boolean.TRUE.toString());
-                copyConfigFolder(fileChanged, serverXmlFileParent, "server.xml");
-                copyFile(fileChanged, serverXmlFileParent, serverDirectory, "server.xml");
+                boolean generateFeaturesSuccess = false;
                 if (generateFeatures) {
                     // custom server.xml modified
-                    incrementGenerateFeatures();
+                    generateFeaturesSuccess = incrementGenerateFeatures();
                 }
+                // suppress install feature warning - property must be set before calling copyConfigFolder
+                System.setProperty(SKIP_BETA_INSTALL_WARNING, Boolean.TRUE.toString());
+                // copyConfigFolder will install features if new ones are detected
+                copyConfigFolder(fileChanged, serverXmlFileParent, "server.xml", generateFeaturesSuccess);
```

Comment:
Even if generate features fails, dev mode should still install user specified features and copy their configuration file to target.

Reply:
So far, in my testing, if generate features fails, then install features fails. It seems better to leave generate features as the error message for users to deal with in this situation. The current design in the UFO is that we do not install features after a failure to generate features. We can revisit this design if necessary.

Reply:
Sounds good, thank you